sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -252,12 +260,12 @@ def sleep(self):
         sleep(self._statement_sleep_seconds)
 
     def refresh_status(self):
-        status = self._http_client.get_session(self.id)[u'state']
+        response = self._http_client.get_session(self.id)
+        status = response[u'state']
+        log_array = response[u'log']
 
-        if status in constants.POSSIBLE_SESSION_STATUS:
-            self.status = status
-        else:
-            raise LivyUnexpectedStatusException(u"Status '{}' not supported by session.".format(status))
```

Comment:
Agreed with @msftristew. This library is very dependent on Livy states. Seeing one we don't know how to handle will have unknown behavior. If removing the check, we need to have a full code review, like he stated.